File: src/server/gkill/api/req_res/update_mi_request.go
package req_res

import "github.com/mt3hr/gkill/src/app/gkill/dao/reps"

type UpdateMiRequest struct {
	SessionID string `json:"session_id"`

	Mi reps.Mi `json:"mi"`

	TXID *string `json:"tx_id"`

	LocaleName string `json:"locale_name"`

	UpdatedKyou *reps.Kyou `json:"updated_kyou"`

	WantResponseKyou bool `json:"want_response_kyou"`
}

File: src/client/classes/api/req_res/update-mi-request.ts
'use strict'

import { Mi } from '@/classes/datas/mi'
import { GkillAPIRequest } from '../gkill-api-request'
import { Kyou } from '@/classes/datas/kyou'

export class UpdateMiRequest extends GkillAPIRequest {

    mi: Mi

    tx_id: string | null = null

    want_response_kyou: boolean

    updated_kyou: Kyou | null | null = null

    constructor() {
        super()
        this.mi = new Mi()
        this.want_response_kyou = false
    }

}


